src/tasks/phase2/database_connection.py

```diff
@@ -0,0 +1,161 @@
+        connection = self.get_db_connection()
+        cursor = connection.cursor()
+        subject_id_list = []
+        for subject_id in subject_list:
+            cursor.execute("SELECT imagename from {0} natural join metadata where id={1};".format(tablename,
+                                                                                                  subject_id[0]))
+            result_column=cursor.fetchall()
+            image_score_tuple_list+=[(i[0],subject_id[1]) for i in result_column]
+            subject_id_list+=[subject_id[0] for i in result_column]
+
+        return image_score_tuple_list,subject_id_list
+
+    def get_object_feature_matrix_from_db(self, tablename, label=None, label_type=None):
+        """
+        Returns the object feature(Data) matrix for dimensionality reduction for a model
+        Output Shape: [# of images in db, feature length]
+        """
+        connection = self.get_db_connection()
+        cursor = connection.cursor()
+        if not label_type:
+            query = "SELECT * from {0}".format(tablename)
+        elif label_type == "aspect":
+            query = "SELECT * FROM {0} WHERE imagename IN  (SELECT imagename FROM metadata WHERE aspectofhand " \
+                    "LIKE '%{1}%')".format(tablename, label)
+        elif label_type == "gender":
+            query = "SELECT * FROM {0} WHERE imagename IN  (SELECT imagename FROM metadata WHERE gender = '{1}')".\
+                format(tablename, label)
+        elif label_type == "accessories":
+            query = "SELECT * FROM {0} WHERE imagename IN  (SELECT imagename FROM metadata WHERE accessories = '{1}')".\
+                format(tablename, label)
+        elif label_type == "subject":
+            query = "SELECT * FROM {0} WHERE imagename IN  (SELECT imagename FROM metadata WHERE id = '{1}')".\
+                format(tablename, label)
+        cursor.execute(query)
+        db_output = cursor.fetchall()
+        obj_feature_matrix = []
+        images = []
+        for image_row in db_output:
+            images.append(image_row[0])
+            obj_feature_matrix.extend(pickle.loads(image_row[1]))
+        return {"images": images, "data_matrix": np.array(obj_feature_matrix)}
+
+    def get_correct_labels_for_given_images(self, image_names = None, label_type=None):
+        conn = self.get_db_connection()
+        cursor = conn.cursor()
+        tablename = 'metadata'
+        if not image_names:
+            query = "SELECT imagename, {1} FROM {0}".format(tablename, self.label_type_map.get(label_type))
+        else:
+            query = "SELECT imagename, {1} FROM {0} WHERE imagename IN {2}".format(tablename, self.label_type_map.
+                                                                                   get(label_type), tuple(image_names))
+        cursor.execute(query)
+        result = cursor.fetchall()
+        return result
+
+    def get_metadata_for_task_8(self,image_names=None):
```

Comment:
do we need this function anymore?

Reply:
We have to check requirement and check if we need this function as it is or we need to modify it.

Reply:
We need to discuss it first.